### ISSUE TITLE
Copy module checks presence of source file on the ansible node instead of the delegated node on ansible 1.9.4

### ISSUE THREAD

**Issue by evrardjp** [ansible/ansible#15929]
##### ISSUE TYPE
- Bug Report
##### ANSIBLE VERSION

```
ansible 1.9.4
  configured module search path = None
```
##### CONFIGURATION

Nothing
##### OS / ENVIRONMENT

DISTRIB_DESCRIPTION="Ubuntu 14.04.4 LTS"
##### SUMMARY

When using ```delegated_to: "{{ inventory_hostname}}" with copy module with ansible 1.9, you could copy file from the delegated node to the delegated node (in the same way as remote_src: with ansible 2.0 and above). However, when doing the copy, the module checks for the presence of the file on the ansible node instead of on the delegated node. 
##### STEPS TO REPRODUCE

The following will fail when /tmp/apt.keys doesn't exist on the ansible node:

```

---
- hosts: aio2
  tasks:
    - shell: apt-key exportall > /tmp/apt.keys
    - copy:
        src: /tmp/apt.keys
        dest: /tmp/apt2.keys
      delegate_to: "{{ inventory_hostname }}"
```
##### EXPECTED RESULTS

Success of the copy to /tmp/apt2.keys, by checking the presence of /tmp/apt.keys on the delegated node
##### ACTUAL RESULTS

```
PLAY [aio2] *******************************************************************

GATHERING FACTS ***************************************************************
<hiddenIP> ESTABLISH CONNECTION FOR USER: root
<hiddenIP> REMOTE_MODULE setup
<hiddenIP> EXEC ssh -C -tt -vvv -o ControlMaster=auto -o ControlPersist=60s -o ControlPath="/root/.ansible/cp/ansible-ssh-%h-%p-%r" -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o ConnectTimeout=10 hiddenIP /bin/sh -c 'mkdir -p $HOME/.ansible/tmp/ansible-tmp-1463685993.63-55612537263382 && echo $HOME/.ansible/tmp/ansible-tmp-1463685993.63-55612537263382'
<hiddenIP> PUT /tmp/tmphxoVl6 TO /root/.ansible/tmp/ansible-tmp-1463685993.63-55612537263382/setup
<hiddenIP> EXEC ssh -C -tt -vvv -o ControlMaster=auto -o ControlPersist=60s -o ControlPath="/root/.ansible/cp/ansible-ssh-%h-%p-%r" -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o ConnectTimeout=10 hiddenIP /bin/sh -c 'LANG=en_US.UTF-8 LC_CTYPE=en_US.UTF-8 /usr/bin/python /root/.ansible/tmp/ansible-tmp-1463685993.63-55612537263382/setup; rm -rf /root/.ansible/tmp/ansible-tmp-1463685993.63-55612537263382/ >/dev/null 2>&1'
ok: [aio2]

TASK: [shell apt-key exportall > /tmp/apt.keys] *******************************
<hiddenIP> ESTABLISH CONNECTION FOR USER: root
<hiddenIP> REMOTE_MODULE command apt-key exportall > /tmp/apt.keys #USE_SHELL
<hiddenIP> EXEC ssh -C -tt -vvv -o ControlMaster=auto -o ControlPersist=60s -o ControlPath="/root/.ansible/cp/ansible-ssh-%h-%p-%r" -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o ConnectTimeout=10 hiddenIP /bin/sh -c 'mkdir -p $HOME/.ansible/tmp/ansible-tmp-1463685995.57-72842283363455 && echo $HOME/.ansible/tmp/ansible-tmp-1463685995.57-72842283363455'
<hiddenIP> PUT /tmp/tmptB3hwK TO /root/.ansible/tmp/ansible-tmp-1463685995.57-72842283363455/command
<hiddenIP> EXEC ssh -C -tt -vvv -o ControlMaster=auto -o ControlPersist=60s -o ControlPath="/root/.ansible/cp/ansible-ssh-%h-%p-%r" -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o ConnectTimeout=10 hiddenIP /bin/sh -c 'LANG=en_US.UTF-8 LC_CTYPE=en_US.UTF-8 /usr/bin/python /root/.ansible/tmp/ansible-tmp-1463685995.57-72842283363455/command; rm -rf /root/.ansible/tmp/ansible-tmp-1463685995.57-72842283363455/ >/dev/null 2>&1'
changed: [aio2] => {"changed": true, "cmd": "apt-key exportall > /tmp/apt.keys", "delta": "0:00:00.045764", "end": "2016-05-19 19:26:35.703611", "rc": 0, "start": "2016-05-19 19:26:35.657847", "stderr": "", "stdout": "", "warnings": []}

TASK: [copy ] *****************************************************************
<hiddenIP> ESTABLISH CONNECTION FOR USER: root
failed: [aio2 -> aio2] => {"failed": true}
msg: could not find src=/tmp/apt.keys

FATAL: all hosts have already failed -- aborting
```

**Comment by bcoca** [comment]
# Possible Misunderstanding

Hi!

Thanks very much for your submission to Ansible.  It sincerely means a lot to us.

We believe the ticket you have filed is being somewhat misunderstood, as one thing works a little differently than stated.

Delegation changes the target of the copy, not the source.

In the future, this might be a topic more well suited for the user list, which you can also post here if you'd like some more help with the above.
- https://groups.google.com/forum/#!forum/ansible-project

Thank you once again for this and your interest in Ansible!

**Comment by evrardjp** [comment]
We agreed on the delegation that changes the target of the copy.
It's not the point.

We could trick ansible 1.9.4 to delegate to the node itself to copy from the node to the node, which would be the same as doing a shell: cp .

However, when doing this ansible still checks that the file is present on the ansible host before copying it to remote node. It completely makes sense, but isn't what the user would expect by default.

Anyway it's fixed by the presence of remote_src in Ansible 2, and I don't think it's gonna be fixed in 1.9 anyway, so we should probably close.

Please take the time to really pay attention to the bug in the future. :p
